web/ajax/event.php

```diff
@@ -167,6 +167,36 @@
       ajaxResponse(array('refreshEvent'=>false, 'refreshParent'=>true));
     }
     break;
+  case 'getselectedtags' :
+    $sql = '
+      SELECT
+        T.*
+      FROM Tags
+        AS T
+      INNER JOIN Events_Tags
+        AS ET
+        ON ET.TagId = T.Id
+      WHERE ET.EventId = ?
+    ';
+    $values = array($_REQUEST['id']);
+    $response = dbFetchAll($sql, NULL, $values);
+    ajaxResponse(array('response'=>$response));
+    break;
+  case 'addtag' :
+    $sql = 'INSERT INTO Events_Tags (TagId, EventId, AssignedBy) VALUES (?, ?, ?)';
+    $values = array($_REQUEST['tid'], $_REQUEST['id'], $user->Id());
+    $response = dbFetchAll($sql, NULL, $values);
+
+    $sql = 'UPDATE Tags SET LastAssignedDate = NOW() WHERE Id = ?';
+    $values = array($_REQUEST['tid']);
+    dbFetchAll($sql, NULL, $values);
+
+    ajaxResponse(array('response'=>$response));
+    break;
+  case 'removetag' :
+    dbFetchAll('DELETE FROM Events_Tags WHERE TagId = ? AND EventId = ?', NULL, array($_REQUEST['tid'], $_REQUEST['id']));
```

Comment:
Why dbFetchAll instead of just dbQuery?

Reply:
I was having trouble with dbQuery while I was trying to figure out the code.  It may work just as well in this case.

Reply:
Changed from dbFetchAll to dbQuery for remove tags.